### DISCUSSION TITLE
5-new-mvc-features-in-dotnet-7/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#91] [Announcements]
# 5-new-mvc-features-in-dotnet-7/

In this post I look at some of the new features introduced for MVC controllers in .NET 7

https://andrewlock.net/5-new-mvc-features-in-dotnet-7/

**Comment by MichalBrylka** [comment]
this does not seem right:
```cs
Enumerable.Range(days.From, days.To)
```
second parameter is `count` so this should be something like
```cs
Enumerable.Range(days.From, days.To - days.From)
```

**Reply by andrewlock** [reply]
Don't think about it to hard, that's tangential to the point of the post 😉 you also need to check that To is larger than From etc etc

**Reply by MichalBrylka** [reply]
I totally agree, some other minor quirks (also tangential to the point) occur here but I just point out bad use of `Enumerable.Range` whenever possible - it's an error in sooo many code reviews I do. It "generally works" when you start from 0 ;-)

**Reply by MichalBrylka** [reply]
other than that - fabulous as always :)

**Comment by arthg** [comment]
Re `ModelMetadataDetailsProvider`

https://github.com/dotnet/aspnetcore/issues/43524

specifically: https://github.com/dotnet/aspnetcore/issues/43524#issuecomment-1227541995

"The SystemTextJsonValidationMetadataProvider provides a ValidationMetadataName metadata that is used during the validation process and not during the ProblemDetails serialization"
